composer/metrics/nlp.py

```diff
@@ -266,9 +299,8 @@ def replace_underscore(text: str) -> str:
 
         return white_space_fix(remove_articles(handle_punc(lower(replace_underscore(answer))))).strip()
 
-    def update(self, outputs: List[str], labels: List[List[str]], batch: Optional[Dict[str, Any]] = None):
-        if batch is None:
-            batch = {}
+    # TODO: rearrange these? why could batch be optional?
```

Comment:
@dakinggg what do you think about making this change now? everywhere else has batch first

Reply:
wow thats annoying. hm...technically thats breaking, but maybe making the signature correct with the base class is more important and its unlikely to actually break anyone... @mvpatel2000 wdyt?

Reply:
@maxisawesome I vote we leave as is for now, and when we actually remove `output_logits`, we make the breaking change here too so that it matches the base class signature. and then call it out in the release notes.

Reply:
Okay, that works